packages/suite/src/components/suite/modals/ReviewTransaction/components/OutputElement.tsx

```diff
@@ -123,63 +131,105 @@ const TruncateWrapper = ({
     children?: React.ReactNode;
 }) => (condition ? <Truncate>{children}</Truncate> : <>{children}</>);
 
-const OutputLine = ({
+const getFingerprint = (
+    tokens: Account['tokens'],
+    symbol: string | undefined,
+): string | undefined => {
+    if (!tokens) {
+        return undefined;
+    }
+
+    const token = tokens.find(token => token.symbol?.toLowerCase() === symbol?.toLowerCase());
+
+    return token?.name;
+};
+
+const OutputElement = ({
     indicator,
     lines,
+    token,
     cryptoSymbol,
     fiatSymbol,
     hasExpansion = false,
     fiatVisible = false,
-}: Props) => (
-    <OutputWrapper>
-        <OutputLeft>
-            {lines.length > 1 ? (
-                <MultiIndicatorWrapper linesCount={lines.length - 1}>
-                    {indicator}
-                </MultiIndicatorWrapper>
-            ) : (
-                <>{indicator}</>
-            )}
-        </OutputLeft>
-        <OutputRight>
-            {lines.map(line => (
-                <OutputRightLine key={line.id}>
-                    <OutputHeadline>
-                        <Truncate>{line.label}</Truncate>
-                    </OutputHeadline>
-                    <OutputValue>
-                        <TruncateWrapper condition={hasExpansion}>
-                            <OutputValueWrapper>
-                                {line.plainValue ? (
-                                    line.value
-                                ) : (
-                                    <FormattedCryptoAmount
-                                        disableHiddenPlaceholder
-                                        value={line.value}
-                                        symbol={cryptoSymbol}
-                                    />
-                                )}
-                            </OutputValueWrapper>
-                            {fiatVisible && (
-                                <>
-                                    <DotSeparatorWrapper>
-                                        <DotSeparator />
-                                    </DotSeparatorWrapper>
-                                    <OutputValueWrapper>
-                                        <FiatValue
+}: Props) => {
+    const { account } = useSelector(state => state.wallet.selectedAccount);
```

Comment:
there is actually "account" available in a relatively near parent component `OutputList` in `components/suite/modals/ReviewTransaction/components/OutputList.tsx`

It goes like this

```
OutputList (has account) -> Output -> OutputElement (takes account using useSelector)
```
So my question is, if it is maybe generally better practice to pass data as prop in case when data is relatively near in the component tree? I am not sure myself. 

Also, maybe it would be nice if this component could stay "dump". maybe even `cardanoFingerprint` could be passed from parent component?

Reply:
✓